xrpl/models/transactions/deposit_preauth.py

```diff
@@ -39,14 +41,85 @@ class DepositPreauth(Transaction):
         init=False,
     )
 
+    authorize_credentials: Optional[List[Credential]] = None
+    """The credential(s) that received the preauthorization. (Any account with these
+    credentials can send preauthorized payments)."""
+
+    unauthorize_credentials: Optional[List[Credential]] = None
+    """The credential(s) whose preauthorization should be revoked."""
+
     def _get_errors(self: Self) -> Dict[str, str]:
         errors = super()._get_errors()
-        if self.authorize and self.unauthorize:
-            errors[
-                "DepositPreauth"
-            ] = "One of authorize and unauthorize must be set, not both."
 
-        if not self.authorize and not self.unauthorize:
-            errors["DepositPreauth"] = "One of authorize and unauthorize must be set."
+        if (
+            self.authorize is None
+            and self.unauthorize is None
+            and self.authorize_credentials is None
+            and self.unauthorize_credentials is None
+        ):
+            errors["DepositPreauth"] = (
+                "Exactly one input parameter amongst authorize, unauthorize, "
+                + "authorize_credentials or unauthorize_credentials must be set. It is "
+                + "invalid if none of the params are specified."
+            )
+
+        if (
+            sum(
+                [
+                    param is not None
+                    for param in (
+                        self.authorize,
+                        self.unauthorize,
+                        self.authorize_credentials,
+                        self.unauthorize_credentials,
+                    )
+                ]
+            )
+            > 1
+        ):
+            errors["DepositPreauth"] = (
+                "More than one input param cannot be specified for DepositPreauth "
+                + "transaction. Please specify exactly one input parameter. "
+            )
+
+        def _validate_credentials_length(
+            credentials: List[Credential], field_name: str
+        ) -> None:
+            if len(credentials) == 0:
+                errors["DepositPreauth"] = f"{field_name} list cannot be empty. "
+            if len(credentials) > 8:
+                errors[
+                    "DepositPreauth"
+                ] = f"{field_name} list cannot have more than 8 elements. "
+
+        # Then replace the checks with:
+        if self.authorize_credentials is not None:
+            _validate_credentials_length(
+                self.authorize_credentials, "AuthorizeCredentials"
+            )
+
+        if self.unauthorize_credentials is not None:
+            _validate_credentials_length(
+                self.unauthorize_credentials, "UnauthorizeCredentials"
+            )
+
+        # Note: Other validity checks like sufficient account-reserve balance,
+        # existence of the issuer, etc on the list of credentials need access to the
+        # blockchain state.
```

Comment:
This comment is unnecessary.

Reply:
fixed in dfd9381b1bbee70da5c186138e1f0d96237ba7ff